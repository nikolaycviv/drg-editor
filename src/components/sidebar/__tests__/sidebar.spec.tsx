import { act, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SideBar from '../sidebar';

describe('Sidebar', () => {
  const tabs = [{ name: 'Resources', icon: 'assets/resources.webp' }];

  it('should render correctly', () => {
    const onClick = vi.fn();

    const tree = renderer
      .create(<SideBar onClick={onClick} tabs={tabs} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should handle tab click', () => {
    const onClick = vi.fn();
    render(<SideBar onClick={onClick} tabs={tabs} />);

    act(() => screen.getByText('Resources').click());

    expect(onClick).toBeCalledWith('Resources');
  });
});
