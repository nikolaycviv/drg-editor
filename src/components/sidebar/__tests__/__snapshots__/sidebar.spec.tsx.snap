// Vitest Snapshot v1

exports[`Sidebar > should render correctly 1`] = `
<div
  className="flex max-w-7xl flex-col md:flex-col lg:flex-row"
>
  <div
    className="mr-3 h-full w-auto border-drg-primary-500 p-3 lg:w-[250px] lg:border-r-2"
  >
    <ul
      className="menu mr-4 flex-row flex-wrap justify-evenly md:flex-row lg:flex-col"
    >
      <li
        className="hover:border-drg-primary-300 display-block my-1 rounded-md font-medium"
        role="tab"
      >
        <button
          className="border-b-4 !border-drg-primary-500 flex-col justify-center rounded-none focus:bg-transparent active:bg-slate-200 active:bg-opacity-10 md:flex-col md:justify-start lg:flex-row"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Resources"
            className="w-12"
            src="assets/resources.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            Resources
          </p>
        </button>
      </li>
    </ul>
  </div>
  <div
    className="box-border flex h-full w-auto flex-row p-6 lg:h-[700px] lg:w-[1200px]"
  />
</div>
`;
